softioc/asyncio_dispatcher.py

```diff
@@ -47,6 +48,25 @@ def close(self):
             self.__atexit = None
 
         self.__shutdown()
+    
+    def wait_for_quit(self):
+
+        async_stop_event = asyncio.Event()
+        stop_event = threading.Event()
+
+        def signal_exit():
+            self.loop.call_soon_threadsafe(async_stop_event.set)
+        
+        async def async_signal_exit():
+            await async_stop_event.wait()
+            stop_event.set()
+
+        for sig in ('SIGINT', 'SIGTERM'):
+            signal.signal(getattr(signal, sig), lambda signum, frame: signal_exit())
+
+        asyncio.run_coroutine_threadsafe(async_signal_exit(), self.loop)
+
+        stop_event.wait()
```

Comment:
I have an alternative suggestion that I think is simpler; it only uses a single Event, and uses `create_task` to allow placing an async task into the async processing loop without actually needing to be in an async function.

```suggestion
        def wait_for_quit(self):

        stop_event = threading.Event()

        async def async_signal_exit():
            stop_event.set()

        for sig in ('SIGINT', 'SIGTERM'):
            signal.signal(getattr(signal, sig), lambda signum, frame: asyncio.create_task(async_signal_exit()))

        stop_event.wait()
```

Reply:
I've tested this proposal. But I get this error message when I close the program. 

```
python test.py 
INFO: PVXS QSRV2 is loaded, permitted, and ENABLED.
Starting iocInit
############################################################################
## EPICS 7.0.7.0
## Rev. 7.0.7.99.0.2
## Rev. Date 7.0.7.99.0.2
############################################################################
iocRun: All initialization complete
^CTraceback (most recent call last):
  File "/home/xavier/test.py", line 29, in <module>
    softioc.non_interactive_ioc()
  File "/home/xavier/test_softioc/lib64/python3.12/site-packages/softioc/softioc.py", line 365, in non_interactive_ioc
    device.dispatcher.wait_for_quit()
  File "/home/xavier/test_softioc/lib64/python3.12/site-packages/softioc/asyncio_dispatcher.py", line 61, in wait_for_quit
    stop_event.wait()
  File "/usr/lib64/python3.12/threading.py", line 655, in wait
    signaled = self._cond.wait(timeout)
               ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.12/threading.py", line 355, in wait
    waiter.acquire()
  File "/home/xavier/test_softioc/lib64/python3.12/site-packages/softioc/asyncio_dispatcher.py", line 59, in <lambda>
    signal.signal(getattr(signal, sig), lambda signum, frame: asyncio.create_task(async_signal_exit()))
                                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.12/asyncio/tasks.py", line 417, in create_task
    loop = events.get_running_loop()
           ^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: no running event loop
sys:1: RuntimeWarning: coroutine 'AsyncioDispatcher.wait_for_quit.<locals>.async_signal_exit' was never awaited
```

Reply:
I added the function to the documentation. And I've modified the `CHANGELOG.rst` file.